### PR TITLE
chore: move files to comply with package layout conventions

### PR DESCRIPTION
**Pull request by renancaraujo**
## Description
Move files to comply with dart [package layout conventions.](https://dart.dev/tools/pub/package-layout#implementation-files)

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

### CODE REVIEW

File: lib/src/callback_debouncer.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+/// {@template callback_debouncer}
+/// A model used for debouncing callbacks.
+///
+/// Is only used internally and should not be used explicitly.
+/// {@endtemplate}
+class CallbackDebouncer {
```
Comment (renancaraujo) [inline lib/src/callback_debouncer.dart:10]: Since this file is not exported. No need to make this class visible for testing anymore.

**Review by felangel — DISMISSED** [review]

**Review by felangel — APPROVED** [review]

File: lib/src/infinite_list.dart [hunk 2]
```diff
@@ -160,13 +160,15 @@ class InfiniteList extends StatefulWidget {
   final ItemBuilder itemBuilder;
 
   @override
-  _InfiniteListState createState() => _InfiniteListState();
+  InfiniteListState createState() => InfiniteListState();
```
Comment (felangel) [inline lib/src/infinite_list.dart:163]: Can't this be `State<_InfiniteListState>`?

**Review by felangel — CHANGES_REQUESTED** [review]
see comment

File: lib/src/infinite_list.dart (hunk 2 above)
Comment (renancaraujo) [reply lib/src/infinite_list.dart:163]: Holy heck... 😂

**Review by felangel — APPROVED** [review]

**Pull request merged**